packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3217,14 +3217,14 @@ export class ContainerRuntime
 			if (runGC) {
 				await this.collectGarbage(
 					{ logger: summaryLogger, runSweep, fullGC },
-					telemetryContext,
+					telemetryContext as ITelemetryContext,
```

Comment:
@markfields @chentong7 what do you think about adding `get()` to `ITelemetryContextExt` but removing it from `ITelemetryContext` (we already removed the other deprecated one, `serialize()`, so why not go all the way to clean up the alpha interface), making the `TelemetryContext` class implement _both_ interfaces (I think we could even use the same set/setMultiple methods to cover both signatures since the non-ext property types are subsets of the new ones, so receiving one of those for `set`-ing should always work), and inside [`incrementTelemetryMetric()`](https://github.com/microsoft/FluidFramework/blob/fcfbf7707d5407927533f9b4da1364c03d0125a6/packages/dds/shared-object-base/src/sharedObject.ts#L763) (the only place that uses `ITelemetryContext.get()`), we confirm that the `get` property exists on the passed-in `telemetryContext`, cast it to `ITelemetryContextExt`, and use it?

Reply:
In fact, not even have `get()` in the Ext interface but only on the `TelemetryContext` class, and `incrementTelemetryMetric` would cast to the class after checking that `get` exists.

Reply:
I think the idea was keep the public interface only have the set function and let implementation type to customize the get function. I will say keep `ITelemetryContext` only have set(), and move get() and serialize() from `TelemetryContext` to `ITelemetryContextExt`

Reply:
My point is that we don't need `get()` or `serialize()` in _either_ of the interfaces. I assumed right now we didn't remove it from `ITelemetryContext` because there is a code path that receives that interface and needs to call `get` on it. We definitely still need `get()` on the _class_ to support that code path, but if we remove it from both interfaces, then I think we can avoid the need for casting here, and we can just cast in a much safer way only inside that single function that needs `get()`. It will still receive an object typed as `ITelemetryContext`, but it can check if the object has a `get` property, and if so cast it to the _class_ `TelemetryContext` and use the function. That would keep both interfaces "clean" in the sense that they would only have set/setMultiple.

Reply:
Yeah I agree with what Alex is outlining here

Reply:
Gotcha. Updating accordingly